File: 백준/레벨Bronze II/2577. 숫자의 개수/숫자의 개수.cc
#include <bits/stdc++.h>
using namespace std;

int digits[10];
int main(void) {
  ios::sync_with_stdio(0);
  cin.tie(0);
  
  int A, B, C;
  cin >> A >> B >> C;

  int number = A * B * C;

  while (number) {
    digits[number % 10]++;
    number /= 10;
  }

  for (int count : digits)
    cout << count << '\n';
}

File: 백준/레벨Bronze II/2577. 숫자의 개수/숫자의 개수.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int A = Integer.parseInt(br.readLine());
        int B = Integer.parseInt(br.readLine());
        int C = Integer.parseInt(br.readLine());
        
        int[] digits = new int[10];
        int number = A * B * C;

        while (number > 0) {
            digits[number % 10]++;
            number /= 10;
        }
        
        StringBuilder sb = new StringBuilder();
        for (int count : digits) {
            sb.append(count).append("\n");
        }
        
        System.out.println(sb);
    }
}